zone/quest_parser_collection.cpp

```diff
@@ -27,1281 +27,1265 @@
-		return iter->second->BotHasQuestSub(evt);
-	}
-
-	return false;
-}
+	std::vector<std::any>* extra_pointers
+)
+{
+	int ret = 0;
 
-bool QuestParserCollection::BotHasQuestSubGlobal(QuestEventID evt) {
-	if (_global_bot_quest_status == QuestUnloaded) {
-		std::string filename;
-		QuestInterface *qi = GetQIByGlobalBotQuest(filename);
-		if (qi) {
-			_global_bot_quest_status = qi->GetIdentifier();
-			qi->LoadGlobalBotScript(filename);
-			return qi->GlobalBotHasQuestSub(evt);
+	for (const auto& e: _load_precedence) {
+		int i = e->DispatchEventBot(event_id, bot, init, data, extra_data, extra_pointers);
+		if (i != 0) {
+			ret = i;
 		}
-	} else if (_global_bot_quest_status != QuestFailedToLoad) {
-		auto iter = _interfaces.find(_global_bot_quest_status);
-		return iter->second->GlobalBotHasQuestSub(evt);
 	}
 
-	return false;
-}
-
-bool QuestParserCollection::BotHasQuestSub(QuestEventID evt) {
-	return BotHasQuestSubLocal(evt) || BotHasQuestSubGlobal(evt);
+	return ret;
 }
 
-QuestInterface *QuestParserCollection::GetQIByBotQuest(std::string &filename) {
-	if (!zone || !zone->IsLoaded()) {
-		return nullptr;
-	}
-
-	// first look for /quests/zone/bot_v[instance_version].ext (precedence)
-	filename = fmt::format("{}/{}/bot_v{}", path.GetQuestsPath(), zone->GetShortName(), zone->GetInstanceVersion());
-	std::string tmp;
-	FILE *f = nullptr;
-
-	auto iter = _load_precedence.begin();
-	while (iter != _load_precedence.end()) {
-		auto ext = _extensions.find((*iter)->GetIdentifier());
-
-		tmp = fmt::format("{}.{}", filename, ext->second);
-
-		f = fopen(tmp.c_str(), "r");
-		if (f) {
-			fclose(f);
-			filename = tmp;
-			return (*iter);
-		}
-
-		++iter;
-	}
-
-	// second look for /quests/zone/bot.ext (precedence)
-	filename = fmt::format("{}/{}/bot", path.GetQuestsPath(), zone->GetShortName());
-
-	iter = _load_precedence.begin();
-	while(iter != _load_precedence.end()) {
-		auto ext = _extensions.find((*iter)->GetIdentifier());
-
-		tmp = fmt::format("{}.{}", filename, ext->second);
-
-		f = fopen(tmp.c_str(), "r");
-		if (f) {
-			fclose(f);
-			filename = tmp;
-			return (*iter);
-		}
-
-		++iter;
+void QuestParserCollection::LoadPerlEventExportSettings(PerlEventExportSettings* s)
+{
+	for (int i = 0; i < _LargestEventID; i++) {
+		s[i].qglobals        = 1;
+		s[i].mob             = 1;
+		s[i].zone            = 1;
+		s[i].item            = 1;
+		s[i].event_variables = 1;
 	}
 
-	// third look for /quests/global/bot.ext (precedence)
-	filename = fmt::format("{}/{}/bot", path.GetQuestsPath(), QUEST_GLOBAL_DIRECTORY);
-	iter = _load_precedence.begin();
-	while (iter != _load_precedence.end()) {
-		auto ext = _extensions.find((*iter)->GetIdentifier());
-
-		tmp = fmt::format("{}.{}", filename, ext->second);
-
-		f = fopen(tmp.c_str(), "r");
-		if (f) {
-			fclose(f);
-			filename = tmp;
-			return (*iter);
-		}
+	const auto& l = PerlEventExportSettingsRepository::All(database);
 
-		++iter;
+	if (l.empty()) {
+		return;
 	}
```

Comment:
Can discard the empty so we can see the loaded message when there's `0` entries

Reply:
Sounds good.